app/src/main/java/com/example/cookbook/data/repositories_impl/RecipeRepositoryImpl.kt

```diff
@@ -21,15 +22,16 @@ class RecipeRepositoryImpl @Inject constructor(
 ) : RecipeRepository {
 
     override fun refreshDatabaseWithRandomRecipes(): Completable {
-        return service.getRandomRecipeList().flatMapCompletable {
-            val list = it.hits?.map { response ->
-                responseToDataMapper(response)
-            } ?: throw Exception(ERROR_MESSAGE)
-            val listOfEntities = list.map { recipe ->
-                dataToEntityMapper(recipe)
+        return Completable.fromAction {
+            val list = service.getRandomRecipeList().map { listResponse ->
+                listResponse.hits?.map { response ->
+                    responseToDataMapper(response)
+                }?.map { recipe ->
+                    dataToEntityMapper(recipe)
+                } ?: throw Exception(ERROR_MESSAGE)
+            }.subscribe { list ->
```

Comment:
Why you don't added error handling in subscribe block ?
it's good approach to move subscribe on low level (viewmodel or other, not in repository)

Reply:
Also move all operations on background thread

Reply:
ok, fixed it, thanks